src/runtime/step_runner.js

```diff
@@ -8,22 +8,27 @@ const {beginTiming, endTiming} = Time
 
 async function run({attachmentManager, defaultTimeout, scenarioResult, step, stepDefinition, parameterRegistry, world}) {
   beginTiming()
-  const parameters = stepDefinition.getInvocationParameters({scenarioResult, step, parameterRegistry})
-  const timeoutInMilliseconds = stepDefinition.options.timeout || defaultTimeout
-
   let error, result
-  const validCodeLengths = stepDefinition.getValidCodeLengths(parameters)
-  if (_.includes(validCodeLengths, stepDefinition.code.length)) {
-    const data = await UserCodeRunner.run({
-      argsArray: parameters,
-      fn: stepDefinition.code,
-      thisArg: world,
-      timeoutInMilliseconds
-    })
-    error = data.error
-    result = data.result
-  } else {
-    error = stepDefinition.getInvalidCodeLengthMessage(parameters)
+
+  try {
+    const parameters = stepDefinition.getInvocationParameters({scenarioResult, step, parameterRegistry})
```

Comment:
I'd prefer we only have the try/catch around this statement. I like try/catch to be very isolated to prevent it from catching errors we don't want it to.